### PR TITLE
test(bun-build-api): improve tests

### PR DESCRIPTION
**Pull request by aboqasem**
### What does this PR do?

- [X] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

File: packages/bun-types/bun.d.ts [hunk 1]
```diff
@@ -1265,7 +1265,7 @@ declare module "bun" {
   type ModuleFormat = "esm"; // later: "cjs", "iife"
 
   interface BuildConfig {
-    entrypoints: string[]; // list of file path
+    entrypoints: [string, ...string[]]; // list of file path
```
Comment (paperclover) [inline packages/bun-types/bun.d.ts]: this type change may break some uses of the api

insight from @Didas-git:

image

imo we shouldn't do this change.

File: test/bundler/bun-build-api.test.ts [hunk 2]
```diff
@@ -168,36 +154,53 @@ describe("Bun.build", () => {
...
 
-  // test("BuildArtifact properties splitting", async () => {
-  //   Bun.gc(true);
-  //   const x = await Bun.build({
-  //     entrypoints: [join(import.meta.dir, "./fixtures/trivial/index.js")],
-  //     splitting: true,
-  //   });
-  //   expect(x.outputs).toHaveLength(2);
-  //   const [indexBlob, chunkBlob] = x.outputs;
+  test("BuildArtifact properties splitting", async () => {
```
Comment (paperclover) [inline test/bundler/bun-build-api.test.ts:155]: lol why were these commented out.

**Review by paperclover — CHANGES_REQUESTED** [review]
looks good but skeptical about the typedef change

File: packages/bun-types/bun.d.ts (hunk 1 above)
Comment (Didas-git) [reply packages/bun-types/bun.d.ts]: To explain what i meant in the last line:
When it comes to this tuple types tss will do a lot of instanciations wich is pretty much useless if you are not making use of the type to begin with.

_the white theme hurts me tho_

Comment (aboqasem) [reply packages/bun-types/bun.d.ts]: makes sense, thanks for the review! i've reverted the type change

**Review by paperclover — APPROVED** [review]

File: test/bundler/bun-build-api.test.ts (hunk 2 above)
Comment (aboqasem) [reply test/bundler/bun-build-api.test.ts:155]: there's even one more haha: https://github.com/oven-sh/bun/issues/8280#issuecomment-1902430096, i'm just not sure what's the expected there

**Pull request closed without merging**